runtime/drivers/duckdb/olap.go

```diff
@@ -630,6 +630,106 @@ func (c *connection) execWithLimits(parentCtx context.Context, stmt *drivers.Sta
 	return err
 }
 
+// convertToEnum converts a varchar col in table to an enum type.
+// Generally to be used for low cardinality varchar columns although not enforced here.
+func (c *connection) convertToEnum(ctx context.Context, table, col string) error {
+	if !c.config.ExtTableStorage {
+		return fmt.Errorf("`cast_to_enum` is only supported when `external_table_storage` is enabled")
+	}
+	c.logger.Info("convert column to enum", zap.String("table", table), zap.String("col", col))
+
+	version, exist, err := c.tableVersion(table)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return fmt.Errorf("table %q does not exist", table)
+	}
+
+	dbName := dbName(table, version)
+	enum := fmt.Sprintf("%s_enum", col)
+
+	var switchErr error
+	err = c.WithConnection(ctx, 100, true, false, func(ctx, ensuredCtx context.Context, _ *dbsql.Conn) error {
+		// check that atleast one non nil value exists in the column
+		res, err := c.Execute(ctx, &drivers.Statement{Query: fmt.Sprintf("SELECT (SELECT count(%s) FROM %s.default WHERE %s IS NOT NULL) > 0 AS cnt", safeSQLName(col), safeSQLName(dbName), safeSQLName(col))})
+		if err != nil {
+			return err
+		}
+
+		var exists bool
+		if res.Next() {
+			if err := res.Scan(&exists); err != nil {
+				_ = res.Close()
+				return err
+			}
+		}
+		_ = res.Close()
+		if !exists {
+			return fmt.Errorf("column %q can't be converted to enum, has zero non null values", col)
+		}
+
+		// scan current db and current schema
+		res, err = c.Execute(ctx, &drivers.Statement{Query: "SELECT current_database(), current_schema()"})
```

Comment:
wondering, should this and all other statements in this method use ensuredCtx instead ?

Reply:
I was hoping to use `ctx` till we drop the enum so that query can be cancelled if required. Post that we are better off with ignoring query cancellation and completing everything to not end up in some failed state like table exists but type doesn't.

Reply:
ok